pepper-apis/src/main/java/org/broadinstitute/ddp/route/PutFormAnswersRoute.java

```diff
@@ -242,29 +248,43 @@ private FormInstance loadFormInstance(Response response,
                                                  String instanceGuid,
+                                                 FormInstance form,
                                                  LanguageDto preferredLangDto,
                                                  UserActivityInstanceSummary instanceSummary) {
-        // For a parent instance, check if there are any child instances and if those are complete.
-        List<ActivityInstanceDto> childInstances = instanceSummary.getInstancesStream()
+        Map<String, List<ActivityInstanceDto>> childInstances = new HashMap<>();
+        instanceSummary.getInstancesStream()
                 .filter(instance -> instanceGuid.equals(instance.getParentInstanceGuid()))
-                .collect(Collectors.toList());
-        for (var childInstanceDto : childInstances) {
-            if (childInstanceDto.getStatusType() != InstanceStatusType.COMPLETE) {
-                // Child instance is not finished but it might not have required questions, so check it.
-                FormInstance childForm = loadFormInstance(
-                        response, handle, userGuid, operatorGuid,
-                        childInstanceDto.getGuid(), preferredLangDto, instanceSummary);
-                if (!childForm.isComplete()) {
-                    String msg = "Status for instance " + instanceGuid + " cannot be set to COMPLETE because the"
-                            + " question requirements are not met for child instance " + childInstanceDto.getGuid();
-                    LOG.info(msg);
-                    throw ResponseUtil.haltError(response, 422,
-                            new ApiError(ErrorCodes.QUESTION_REQUIREMENTS_NOT_MET, msg));
+                .forEach(instance -> childInstances
+                        .computeIfAbsent(instance.getActivityCode(), key -> new ArrayList<>())
+                        .add(instance));
+        for (FormSection section : form.getAllSections()) {
+            for (FormBlock block : section.getBlocks()) {
+                if (block.getBlockType() != BlockType.ACTIVITY || !block.isShown()) {
+                    continue;
+                }
+                var nestedActivityBlock = (NestedActivityBlock) block;
+                List<ActivityInstanceDto> childInstanceDtos = childInstances
+                        .getOrDefault(nestedActivityBlock.getActivityCode(), new ArrayList<>());
+                for (var childInstanceDto : childInstanceDtos) {
+                    if (childInstanceDto.getStatusType() != InstanceStatusType.COMPLETE) {
+                        // Child instance is not finished but it might not have required questions, so check it.
+                        FormInstance childForm = loadFormInstance(
+                                response, handle, userGuid, operatorGuid,
+                                childInstanceDto.getGuid(), preferredLangDto, instanceSummary);
+                        if (!childForm.isComplete()) {
```

Comment:
Similar isComplete-check and halt error applied for a parent activity. Might be make sense to extract to method? (only a small difference in msg text).

Reply:
I think I will leave it for now.